src/libstd/sys/unix/ext/fs.rs

```diff
@@ -215,36 +215,312 @@ impl OpenOptionsExt for OpenOptions {
 // casts and rely on manual lowering to `stat` if the raw type is desired.
 #[stable(feature = "metadata_ext", since = "1.1.0")]
 pub trait MetadataExt {
+    /// Returns the ID of the device containing the file.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # #[cfg(target_os = "linux")] {
```

Comment:
This is in the `unix` module, not `linux`, so it would be more accurate to use `cfg(unix)` here. (I would prefer not having this wrapper here at all, but if you're going to keep it then that's my suggestion.)